sentry/src/main/java/io/sentry/PropagationContext.java

```diff
@@ -67,18 +71,10 @@ public PropagationContext(final @NotNull PropagationContext propagationContext)
         propagationContext.getTraceId(),
         propagationContext.getSpanId(),
         propagationContext.getParentSpanId(),
-        cloneBaggage(propagationContext.getBaggage()),
```

Comment:
We were cloning baggage here for each scope fork meaning a non frozen baggage would never freeze. This might no longer be a problem after having done a lot more changes now. However I don't see a reason to clone / separate any pre-existing baggage. If anything we should make sure a new one is created when isolation scope is forked.